### PR TITLE
DAOS-13771 control: Unit test TestSupport_getRunningConf() fix

### PR DESCRIPTION
**Pull request by ravalsam**
Removing the unit tests which required fake engine to start, And this test case already part of functional test.

Quick-build: true
Skip-func-test: true

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/control/lib/support/log_test.go [hunk 1]
```diff
@@ -29,9 +29,17 @@ func startFakeEngine(t *testing.T) bool {
 	cmd := exec.Command("bash", "-c", runCmd)
 	err := cmd.Start()
 	if err != nil {
+		t.Logf("FakeEngine Failed to start with err = %s", err.Error())
```
Comment (kjacque) [inline src/control/lib/support/log_test.go:32]: A few items to clean up, while you are refactoring:

- Call `t.Helper()` at the start of the `startFakeEngine` and `killFakeEngine` helper functions. It makes debugging failures a bit easier.
- Call `t.Fatal()` with the log message for the error cases. Right now I don't think any of these will be detected, as the return value isn't checked.
- Remove the boolean return values, as they aren't checked anyway.

**Review by kjacque — CHANGES_REQUESTED** [review]

File: src/control/lib/support/log_test.go (hunk 1 above)
Comment (ravalsam) [reply src/control/lib/support/log_test.go]: Done

**Review by tanabarr — APPROVED** [review]
LGTM, let's get this landed so we can unblock CI

**Comment by tanabarr** [comment]
I would be in favour of landing ASAP to see if we can unblock CI.

**Review by kjacque — APPROVED** [review]
Thanks, I think this will make failures clearer in the future.

**Review by kjacque — APPROVED** [review]

**Review by knard-intel — APPROVED** [review]

**Comment by ashleypittman** [comment]
This PR has 188 valgrind reports which I don't think are known, is that expected?

**Comment by ashleypittman** [comment]
In fact looking at the test results almost every build of this PR has the same 188 errors, I don't think this is ready for landing yet.

**Review by mchaarawi — COMMENTED** [review]
non of the valgrind issues and unit test failures seem to be on master or seen elsewhere. please re-request gatekeeper when this is ready.

**Comment by ravalsam** [comment]
> non of the valgrind issues and unit test failures seem to be on master or seen elsewhere. please re-request gatekeeper when this is ready.

Some of the tests from go code is removed. Not sure how this related to other unit test failure. 
I will check.

**Comment by ravalsam** [comment]
I rebase the master code and ran default tests without any tag and all unit test worked fine. I am not sure if Quick-build: true is making those failures or not.
But this PR seems ready to land as all unit test passed.

**Comment by daltonbohning** [comment]
> I rebase the master code and ran default tests without any tag and all unit test worked fine. I am not sure if Quick-build: true is making those failures or not. But this PR seems ready to land as all unit test passed.

`Quick-build: true` does have problems. I'm not sure if it should cause the unit tests, but I wouldn't be surprised. But either way, `Quick-build: true` shouldn't be used for landing anyway

**Comment by ashleypittman** [comment]
Can you merge this with latest master and then re-push without any commit tags please?

**Comment by ravalsam** [comment]
> Can you merge this with latest master and then re-push without any commit tags please?

Merged with master and all checks looks OK. Please merge this.

**Pull request merged**